fairscale/nn/data_parallel/fully_sharded_data_parallel.py

```diff
@@ -2238,27 +2238,39 @@ def _gather_optim_state(
             # param that has the optimizer state. So we handle it with the correct
             # parameter list.
             non_shared_params = cast(FullyShardedDataParallel, self._fsdp_instances[k]).non_shared_params()
+            non_shared_world_size = self._fsdp_instances[k].world_size
+            non_shared_process_group = self._fsdp_instances[k].process_group
+            # This is the world size and process group of the FSDP submodule which can be
+            # different than the parent module.
+            if non_shared_world_size != self.world_size:
+                world_size = non_shared_world_size
+                process_group = non_shared_process_group
```

Comment:
`world_size` and `process_group` might be very generic variable names. perhaps rename them to `param_world_size` and `param_process_group`?

Reply:
Modified the code to simplify logic which results in using an existing param name.